src/main/java/seedu/agendum/model/ModelManager.java

```diff
@@ -82,6 +82,7 @@ public ReadOnlyToDoList getToDoList() {
 
     /** Raises an event to indicate the model has changed */
     private void indicateToDoListChanged() {
+        toDoList.resetData(toDoList);
```

Comment:
Correct example: If you add a new floating task, the floating task gets a new index and all the completed task index should increase by 1 as a result.
Without this line: Completed task list won't know that the index is updated since there is no change in the filtered completed task list. (index does not change and there is a duplicate)
This forces it to load the index again